metaflow/plugins/kfp/kfp.py

```diff
@@ -587,6 +591,40 @@ def _set_container_resources(
             if toleration:
                 container_op.add_toleration(toleration)
 
+        if kfp_component.spot_decorator:
+            affinity_match_expressions.append(
+                V1NodeSelectorRequirement(
+                    key="k8s.zg-aip.net/cost-type",
+                    operator="In",
+                    values=["spot"],
+                )
+            )
+
+            # ensures the pod created has the correct toleration corresponding to the taint
+            # on the spot node for it to be scheduled on that node
+            toleration = V1Toleration(
+                # the `effect` parameter must be specified at the top!
+                # otherwise, there is undefined behavior
+                effect="NoSchedule",
+                key="k8s.zg-aip.net/cost-type",
```

Comment:
Any chance we could use a [well known label](https://karpenter.sh/preview/tasks/scheduling/#supported-labels) instead? Looks like the primary one is `karpenter.sh/capacity-type: spot`

Or if we want to decouple things maybe syncing up with ZGCP on what toleration key they using for the spot rollout on their end for their direct customers so we don't need a custom AIP specific label?

Reply:
fixed with using `karpenter.sh/capacity-type`!